src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -278,19 +278,20 @@ def get_vm_cloud_type(self):
         for i in range(0, Constants.MAX_IMDS_CONNECTION_RETRY_COUNT):
             try:
```

Comment:
Add a log in here with text "Connecting to IMDS endpoint..." or "Querying IMDS endpoint..."